lib/manager/terraform/lockfile/index.ts

```diff
@@ -80,40 +80,47 @@ export async function updateArtifacts({
       // update all locks in the file during maintenance --> only update version in constraints
       const maintenanceUpdates = await updateAllLocks(locks);
       updates.push(...maintenanceUpdates);
-    } else if (
-      ['provider', 'required_provider'].includes(updatedDeps[0].depType)
-    ) {
-      // update only specific locks but with constrain updates
-      const dep = updatedDeps[0];
+    } else {
+      const filteredUpdates = updatedDeps.filter((dep) =>
+        ['provider', 'required_provider'].includes(dep.depType)
+      );
+      const lockUpdates = await pMap(
+        filteredUpdates,
+        async (dep) => {
+          const lookupName = dep.lookupName ?? dep.depName;
 
-      const lookupName = dep.lookupName ?? dep.depName;
+          // handle cases like `Telmate/proxmox`
+          const massagedLookupName = lookupName.toLowerCase();
 
-      // handle cases like `Telmate/proxmox`
-      const massagedLookupName = lookupName.toLowerCase();
+          const repository = massagedLookupName.includes('/')
+            ? massagedLookupName
+            : `hashicorp/${massagedLookupName}`;
+          const registryUrl = dep.registryUrls
+            ? dep.registryUrls[0]
+            : TerraformProviderDatasource.defaultRegistryUrls[0];
+          const newConstraint = isPinnedVersion(dep.newValue)
+            ? dep.newVersion
+            : dep.newValue;
+          const updateLock = locks.find(
+            (value) => value.lookupName === repository
+          );
+          const update: ProviderLockUpdate = {
+            newVersion: dep.newVersion,
+            newConstraint,
+            newHashes: await TerraformProviderHash.createHashes(
+              registryUrl,
+              repository,
+              dep.newVersion
+            ),
+            ...updateLock,
+          };
+          return update;
+        },
+        { concurrency: 4 }
```

Comment:
Do we really benefit from this concurrency? We already have it in lower level, so it would multiply. So it can cause high load on terraform  registry. 🤔

Reply:
Good point. I'll remove it.